cluster/v1alpha1/types.go

```diff
@@ -0,0 +1,100 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Cluster"
+
+// ManagedClusterSet is a set of managed clusters.
+type ManagedClusterSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of the desired managed clusters
+	Spec ManagedClusterSetSpec `json:"spec"`
+
+	// Status represents the current status of the cluster set
+	// +optional
+	Status ManagedClusterSetStatus `json:"status,omitempty"`
+}
+
+// ManagedClusterSetSpec describes the attributes of the desired managed clusters
```

Comment:
nit (and you don't have to change this now, I know @cadawson is working on this repo and he seems like he might be interested in these details): should we say ManagedCluster instead of "managed cluster"?

Reply:
Updated.